flowredux/src/commonMain/kotlin/com/freeletics/flowredux/sideeffects/SideEffect.kt

```diff
@@ -15,7 +40,11 @@ internal abstract class SideEffect<InputState : S, S, A> {
 
     abstract val isInState: IsInState<S>
 
-    abstract fun produceState(actions: Flow<Action<A>>, getState: GetState<S>): Flow<ChangedState<S>>
+    abstract fun produceState(getState: GetState<S>): Flow<ChangedState<S>>
+
+    open suspend fun sendState(state: S) {}
+
+    open suspend fun sendAction(action: A) {}
```

Comment:
One thing I don't completely like is that it would allow a side effect implementation to call these. It's only internal so it's not that big of a problem and I couldn't come up with something better.